src/server/enrichment-map/emap/generateCys.js

```
@@ -20,7 +20,24 @@ const _ = require('lodash');
 
 // input ["GO:1902275", "GO:2001252", "GO:1905269", "GO:0051053"]
 // returns a cytoscape object
-const generateCys = (pathwayIdList) => {
+const generateCys = (pathwayIdList, JCOCWeight) => {
+  let JCWeight = Number(JCOCWeight.JCWeight);
+  let OCWeight = Number(JCOCWeight.OCWeight);
+  if (JCWeight < 0 || JCWeight > 1) {
+    return 'ERROR: JCWeight out of range [0, 1]';
+  }
+  if (OCWeight < 0 || OCWeight > 1) {
+    return 'ERROR: OCWeight out of range [0, 1]';
+  }
+  if (!isNaN(OCWeight) && !isNaN(JCWeight) && OCWeight + JCWeight != 1) {
+    return 'ERROR: OCWeight+JCWeight should be 1';
```

throw errors here instead of returning strings